extensions/wikia/UserLogin/UserSignupSpecialController.class.php

```diff
@@ -690,8 +690,18 @@ protected function shouldDisableCaptcha() {
 	 * Track an event with a given label with category 'user-sign-up' and action 'request'
 	 *
 	 * @param string $label
+	 * @return void
 	 */
 	protected function track( $label ) {
+		// Check request headers to make sure this is not a mirrored
+		// request, which could result in duplicate tracking events
+		$headers = getallheaders();
```

Comment:
What do you think about moving this logic into ```\Track::event```? This would avoid problems like this in future.

Also I'm thinking about extracting those lines into new method.